includes/class-wc-payments-payment-request-session.php

```diff
@@ -37,7 +37,6 @@ class WC_Payments_Payment_Request_Session {
 	public function init() {
 		// adding this filter with a higher priority than the session handler of the Store API.
 		add_filter( 'woocommerce_session_handler', [ $this, 'add_payment_request_store_api_session_handler' ], 20 );
-		add_filter( 'rest_post_dispatch', [ $this, 'store_api_headers' ], 10, 3 );
```

Comment:
I moved this as a side-effect of `add_payment_request_store_api_session_handler`, because I figured that `store_api_headers` was doing similar checks on the nonce as `add_payment_request_store_api_session_handler`. So I think it's fine to clear up the `store_api_headers` method from the same checks 🤷 .